Add ImageCleaner functionality

LinqManJr opened the pull request.

LinqManJr committed:
add ImageCleaner functionality

add vue components on client
add routes
add ImageCleanerAdminService on server
add DI ImageCleanerAdminService

LinqManJr committed:
Merge branch 'LinqManJr_imageCleaner' into LinqManJr_dev

DmitrijOkeanij commented on SunEngine.Admin/Services/ImageCleanerAdminService.cs:
```diff
@@ -0,0 +1,173 @@
+
+            foreach (var imagePath in imagesInDirectory)
+            {
+                var path = string.Join("\\", imagePath.Replace('/', '\\').Split('\\').Skip(2));
```
Такой синтаксис нельзя использовать, вместо этого делается Path.Combine()
Причина в том что на разных ОС разные разделители, и пути на другой ОС могут не работать.
Вообщем не просто могут а так и есть, у меня файлы получаются но не стираются на Ubuntu

DmitrijOkeanij commented on SunEngine.Admin/Services/ImageCleanerAdminService.cs:
```diff
@@ -0,0 +1,173 @@
+        private readonly HtmlParser _htmlParser;
+        private string _imagesPath;
+
+        public ImageCleanerAdminService(DataBaseConnection dataBaseConnection, string imagesPath = @"wwwroot/UploadImages")
```
Надо брать название директории из ImagesOptions класса, посмотри как сделано в UploadImagesController
![image](https://user-images.githubusercontent.com/10674646/58711863-06a9cd00-83c8-11e9-9d12-35a4e2a8acf4.png)

DmitrijOkeanij commented on SunEngine.Admin/Controllers/ImagesCleanerController.cs:
```diff
@@ -0,0 +1,35 @@
+        {
+            var imagesPath = await imageCleanerService.GetImageSourcesForCleanAsync();
+                        
+            return Ok(new {Json = string.Join('\n', imagesPath)});            
```
Возвращай сразу массив
`return Ok(imagesPath);`
А то лишний код получается, сначала в строку делаешь а потом обратно.

LinqManJr replied:
> Возвращай сразу массив
> `return Ok(imagesPath);`
> А то лишний код получается, сначала в строку делаешь а потом обратно.

так ничего не отображает тогда

LinqManJr replied:
обошел это, только на клиенте все равно та же процедура

Dmitrij-Polyanin commented:
Спасибо! Классная работа.

DmitrijOkeanij commented on Client/src/modules/admin/ImagesCleaner.vue:
```diff
@@ -0,0 +1,74 @@
+          data: {}
+        })
+        .then(response => {   
+          this.error = null;       
```
`this.error = null;` лишнее

DmitrijOkeanij commented on Client/src/modules/admin/ImagesCleaner.vue:
```diff
@@ -0,0 +1,74 @@
+    return {
+      result : null,
+      json: null,
+      error: null,
```
`error: null,`
лишняя переменная

Dmitrij-Polyanin commented:
Добавь ещё всплывающее сообщение, если нажать обновить картинки.

Dmitrij-Polyanin commented:
Кстати можно что бы изображения не стретчились в квадрат а умещались с сохранением пропорций, но лишнее убиралось, можно вот [здесь](https://stackoverflow.com/questions/11757537/css-image-size-how-to-fill-not-stretch#29103071) посмотреть.
Хотя и так нормально, даже не знаю что лучше. Можно попробовать и посмотреть что более симпатично выглядит.

LinqManJr committed:
previous pr fixes

fixes client code
fixes server-side code (add IOptions<ImagesOptions>) to constructor
fix getting image paths (now use Path.Combine)
fix controller method - GetAllImages
add notification to client after reload click
add disable button if images folder is empty
